integration_test.go

```diff
@@ -86,19 +125,222 @@ func testCartographerInternalState(t *testing.T, svc slam.Service, dataDir strin
+	saveInternalState(t, internalState, dataDir)
+}
+
+func saveInternalState(t *testing.T, internalState []byte, dataDir string) {
 	timeStamp := time.Now()
 	filename := filepath.Join(dataDir, "map", "map_data_"+timeStamp.UTC().Format(dataprocess.SlamTimeFormat)+".pbstream")
-	err = os.WriteFile(filename, internalState, 0o644)
+	err := os.WriteFile(filename, internalState, 0o644)
+	test.That(t, err, test.ShouldBeNil)
+}
+
+func saveInternalStateFullMod(t *testing.T, internalState []byte, dataDir string) {
+	timeStamp := time.Now()
+	internalStateDir := filepath.Join(dataDir, "internal_state")
+	if err := os.Mkdir(internalStateDir, 0o755); err != nil {
+		t.Error("failed to create test internal state directory")
+	}
+	filename := filepath.Join(internalStateDir, "map_data_"+timeStamp.UTC().Format(dataprocess.SlamTimeFormat)+".pbstream")
+	if err := os.WriteFile(filename, internalState, 0o644); err != nil {
+		t.Error("failed to write test internal state")
+	}
+}
+
+func testHelperCartographerFullMod(
+	t *testing.T,
+	dataDirectory string,
+	subAlgo viamcartographer.SubAlgo,
+	logger golog.Logger,
+	replaySensor bool,
+	mapRateSec int,
+	expectedMode cartofacade.SlamMode,
+) []byte {
+	t.Helper()
+	termFunc := internaltesthelper.InitTestCL(t, logger)
+	defer termFunc()
+
+	attrCfg := &vcConfig.Config{
+		Sensors: []string{"stub_lidar"},
+		ConfigParams: map[string]string{
+			"mode": reflect.ValueOf(subAlgo).String(),
+		},
+		MapRateSec:              &mapRateSec,
+		DataDirectory:           dataDirectory,
+		ModularizationV2Enabled: &_true,
+	}
+
+	done := make(chan struct{})
+	sensorReadingInterval := time.Millisecond * 200
+	timedSensor, err := testhelper.FullModIntegrationLidarTimedSensor(t, attrCfg.Sensors[0], replaySensor, sensorReadingInterval, done)
+	test.That(t, err, test.ShouldBeNil)
+
+	svc, err := internaltesthelper.CreateFullModSLAMServiceIntegration(t, attrCfg, timedSensor, logger)
+	test.That(t, err, test.ShouldBeNil)
+	start := time.Now()
+
+	cSvc, ok := svc.(*viamcartographer.CartographerService)
+	test.That(t, ok, test.ShouldBeTrue)
+	test.That(t, cSvc.SlamMode, test.ShouldEqual, expectedMode)
+
+	ctx, cancelFunc := context.WithTimeout(context.Background(), time.Second*5)
+
+	defer cancelFunc()
+
+	// wait till all lidar readings have been read
+	if !utils.SelectContextOrWaitChan(ctx, done) {
+		test.That(t, errors.New("test timeout"), test.ShouldBeNil)
+	}
+
+	testCartographerFullModPosition(t, svc, attrCfg.Sensors[0])
+	testCartographerMap(t, svc, cSvc.SlamMode == cartofacade.LocalizingMode)
+
+	internalState, err := slam.GetInternalStateFull(context.Background(), svc)
+	test.That(t, err, test.ShouldBeNil)
+
+	// Close out slam service
+	test.That(t, svc.Close(context.Background()), test.ShouldBeNil)
+
+	stop := time.Now()
+	testDuration := stop.Sub(start)
+	t.Logf("test duration %d", testDuration)
+
+	// Check that a map was generated as the test has been running for more than the map rate msec
```

Comment:
Should this be asserted before testing that the map was generated?

Reply:
nvm I see it is

Reply:
mind if I resolve?

Reply:
Nope go ahead!